### PR TITLE
Update editorconfig & fix warnings and errors appropriately

### PR DESCRIPTION
**Pull request by vexx32**
## Description Of Changes

- Update editorconfig per team discussions
- Fix any warning or error-level complaints that resulted from this
- Run dotnet format on everything to make sure the whitespace, indentation, etc., are all set

## Motivation and Context

Keep code looking good and consistent.

## Testing

- Run CI tests. There should be no functional changes here.
- Open project files in VS. There will be some Suggestion-level things remaining in many files, but no warnings or errors from the code style rules added/changed.

### Operating Systems Testing

N/A

## Change Types Made

* [ ] Bug fix (non-breaking change).
* [x] Feature / Enhancement (non-breaking change).
* [ ] Breaking change (fix or feature that could cause existing functionality to change).
* [ ] Documentation changes.
* [ ] PowerShell code changes.

## Change Checklist

* [ ] Requires a change to the documentation.
* [ ] Documentation has been updated.
* [ ] Tests to cover my changes, have been added.
* [x] All new and existing tests passed?
* [ ] PowerShell code changes: PowerShell v2 compatibility checked?

## Related Issue

Fixes #

### CODE REVIEW

**Comment by vexx32** [comment]
Kinda iffy on whether this needed an issue opened or not, if y'all want me to open an issue and re-tag stuff against it, let me know?

**Review by gep13 — DISMISSED** [review]
LGTM!

**Comment by gep13** [comment]
> @vexx32 said...
Kinda iffy on whether this needed an issue opened or not, if y'all want me to open an issue and re-tag stuff against it, let me know?

I think I am actually okay with this just going through as (maint) commits.  They are logically broken up per rule that is being worked on, which made it super easy to review this!

Thank you very much for getting this done!

**Comment by gep13** [comment]
@vexx32 looks like at least one unit test is now failing as a result of the changes that you have made. Are you in a position to take a look at this, and get it fixed up?

File: src/chocolatey.tests/infrastructure.app/nuget/ChocolateyNuGetProjectContextSpecs.cs [hunk 1]
```diff
@@ -85,7 +85,7 @@ public void Should_log_info_information_in_child_logger()
             }
 
             [TestCase(LogLevel.Debug)]
```
Comment (gep13) [inline src/chocolatey.tests/infrastructure.app/nuget/ChocolateyNuGetProjectContextSpecs.cs:87]:
```suggestion
            [TestCase(NuGet.Common.LogLevel.Debug)]

```

**Comment by gep13** [comment]
@vexx32 I am going to move forward with merging this PR, as the failing Docker Build is not related to the changes that you have made.

**Comment by gep13** [comment]
@vexx32 thanks again for working through all these rules and fixing up all the required changes!

**Pull request merged**